tests/test_project.py

```diff
@@ -446,30 +447,34 @@ def a(job):
 
         # test setting neither nranks nor omp_num_threads
         for job in project:
-            for next_op in project.next_operations(job):
-                assert next_op.directives['np'] == 1
+            with pytest.deprecated_call():
+                for next_op in project.next_operations(job):
+                    assert next_op.directives['np'] == 1
 
         # test only setting nranks
         for i, job in enumerate(project):
             job.doc.nranks = i+1
-            for next_op in project.next_operations(job):
-                assert next_op.directives['np'] == next_op.directives['nranks']
+            with pytest.deprecated_call():
+                for next_op in project.next_operations(job):
+                    assert next_op.directives['np'] == next_op.directives['nranks']
             del job.doc['nranks']
 
         # test only setting omp_num_threads
         for i, job in enumerate(project):
             job.doc.omp_num_threads = i+1
-            for next_op in project.next_operations(job):
-                assert next_op.directives['np'] == next_op.directives['omp_num_threads']
+            with pytest.deprecated_call():
+                for next_op in project.next_operations(job):
+                    assert next_op.directives['np'] == next_op.directives['omp_num_threads']
             del job.doc['omp_num_threads']
 
         # test setting both nranks and omp_num_threads
         for i, job in enumerate(project):
             job.doc.omp_num_threads = i+1
             job.doc.nranks = i % 3 + 1
             expected_np = (i + 1) * (i % 3 + 1)
-            for next_op in project.next_operations(job):
-                assert next_op.directives['np'] == expected_np
+            with pytest.deprecated_call():
```

Comment:
Just use the `_next_operations` function no need to wrap in `pytest.deprecated_call`.